inorbit_edge/robot.py

```diff
@@ -441,6 +447,49 @@ def publish_odometry(
         msg.speed_available = True
         self.publish_protobuf(MQTT_TOPIC_ODOMETRY, msg)
 
+    def publish_path(self, path_points, path_id="0", ts=None):
+        """Publish robot path
+
+        Send a list of points representing the path the robot
+        is traversing. This method only sends the data to InOrbit
+        for displaying purposes, meaning that the path provided
+        here won't make the robot to move
+
+        Args:
+            path_points (List[Tuple[int. int]]): List of x, y points
+                the robot would go through.
+        """
+
+        if len(path_points) > ROBOT_PATH_POINTS_LIMIT:
+            self.logger.warn(
+                "Path has {} points. Only the first {} points will be used.".format(
+                    len(path_points), ROBOT_PATH_POINTS_LIMIT
+                )
+            )
+
+        # Generate ``PathPoint`` protobuf messages
+        # from the list of path point tuples
+        pb_path_points = []
+        for path_point in path_points[:ROBOT_PATH_POINTS_LIMIT]:
+            pb_path_point = PathPoint()
```

Comment:
can this be done in a more pythonic way, using `map` perhaps ?

Reply:
Actually, we can. Positional arguments are not allowed but named are:

```python
>>> PathPoint(1,2)
Traceback (most recent call last):
  File "<stdin>", line 1, in <module>
TypeError: No positional arguments allowed
>>> PathPoint(x=1,y=2)
x: 1.0
y: 2.0
```